### PR TITLE
Conan support (rudimentary)

### PR DESCRIPTION
**Pull request by JonasProgrammer**
Hi,

sorry this took a little while, but here is finally a rudimentary conan integration.

The package aims to be compatible with both the `cmake_paths` generator, exporting the existing `ozo-config.cmake`, `ozo-targets.cmake` etc. files, as well as `cmake_find_package`, mirroring relevant definitions and dependencies from `CMakeLists.txt` in the conan components.

The reason for this double-specification is, that the central conan repository does not allow cmake config files to be exported, so this is my attempt at keeping it possible to be used in both scenarios. Potential further conan-center compatibility is also why I decided to pull in `resource_pool` as a package itself.

Please note, that this then still does not fully comply with the requirements for `conan-center`, I just tried to keep it in a way, so a migration further down the line is easy. Especially in regards to versioning and releasing versions, there is some more discussion needed, before we can try and get them included there.

### CODE REVIEW

**Comment by JonasProgrammer** [comment]
Well sorry for this build spam, getting everything to run on travis was a little harder than anticipated

File: conanfile.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        version = re.search(r"^\s*project\(ozo\s+VERSION\s+([^\s)]+)", content, re.M).group(1)
+        return version.strip()
+    except Exception:
+        return None
+
+
+class Ozo(ConanFile):
+    name = 'ozo'
+    version = get_version()
+    license = 'MIT'
```
Comment (elsid) [inline conanfile.py]: ozo has PostgreSQL License

**Review by elsid — CHANGES_REQUESTED** [review]
A separate docker image for conan looks like an optimizaiton but this creates complications for the whole setup. I don't see how `docker_conan clang release` will work and `docker clang conan` is also incorrect. But still build.sh help suggest that these commands are correct. Can't we install all conan related packages into the existing build image? Generally the idea for this image is to have it built on your machine for ozo development, update it very rarely and don't remove it. With a separate conan image to test all possible changes locally you'd need two images.

File: conanfile.py (hunk 1 above)
Comment (JonasProgrammer) [reply conanfile.py]: My bad, that's what you get from copying and pasting the recipe instead of writing it from the ground up.

**Comment by JonasProgrammer** [comment]
I made some further changes to incorporate fixes to #280, as the conan package should also add those options conditionally, depending on compiler and compiler version. Looking at [other examples](https://github.com/conan-io/conan-center-index/blob/689e2663fe41689259f61a5dbd3fe245e24c03b9/recipes/mp-units/all/conanfile.py#L64) from `conan-center`, this behaviour seems to be expected.

File: scripts/build.sh [hunk 2]
```diff
@@ -195,9 +223,9 @@ case "$1" in
         build_${1}
         exit 0
     ;;
-    docker)
+    docker|docker_conan)
```
Comment (elsid) [inline scripts/build.sh]: `docker_conan` is not used anymore.

File: scripts/build.sh [hunk 3]
```diff
@@ -195,9 +223,9 @@ case "$1" in
...
         set -x
-        launch_in_docker $2 $3 $4 $5
+        launch_in_${1} $2 $3 $4 $5
```
Comment (elsid) [inline scripts/build.sh]: This change is not required anymore.

**Review by elsid — APPROVED** [review]

**Comment by thed636** [comment]
Guys, thanks for the contribution!

**Pull request merged**